### PR TITLE
Removed the hard exception in Constrained PrepSRL

### LINKED ISSUE #458: Fatal error while evaluating classifier ConstrainedPrepSRLClassifier
Suppose I have the following code for PrepSRL:

```
Preprocessor preprocessor = new Preprocessor(PrepSRLConfigurator.defaults());
Annotator` annotator = new PrepSRLAnnotator();
String longSentence = "That put him a stroke ahead of Curt Byrum, John Adams and " + 
                      "Lonnie Clements , all of whom are looking for their first " + 
                       "victory on the PGA Tour .";
TextAnnotation ta = TextAnnotationUtilities.createFromTokenizedString(longSentence);
preprocessor.annotate(ta);    
View view = annotator.getView(ta); // ERROR HERE!
```
On the last line (line 75 in my code), I get the following error:

```
ILP ERROR: Unsatisfiable constraints!
LBJava ERROR: Fatal error while evaluating classifier ConstrainedPrepSRLClassifier: edu.illinois.cs.cogcomp.lbjava.infer.InferenceNotOptimalException: Failed to solve inference problem with the following head object: ahead of
edu.illinois.cs.cogcomp.lbjava.infer.InferenceNotOptimalException: Failed to solve inference problem with the following head object: ahead of
	at edu.illinois.cs.cogcomp.lbjava.infer.ILPInference.infer(ILPInference.java:237)
	at edu.illinois.cs.cogcomp.lbjava.infer.ILPInference.valueOf(ILPInference.java:276)
	at edu.illinois.cs.cogcomp.prepsrl.inference.ConstrainedPrepSRLClassifier.discreteValue(ConstrainedPrepSRLClassifier.java:76)
	at edu.illinois.cs.cogcomp.prepsrl.PrepSRLAnnotator.addView(PrepSRLAnnotator.java:85)
	at edu.illinois.cs.cogcomp.annotation.Annotator.lazyAddView(Annotator.java:181)
	at edu.illinois.cs.cogcomp.annotation.Annotator.getView(Annotator.java:166)
	at Demo.main(Demo.java:75)
```
How can this be fixed? I tried catching both the `InferenceNotOptimalException` and the generic `Exception` (quick dirty fixes), but that didn't work out.

### PR DESCRIPTION
**Pull request by christos-c**
Fixes #458 
Replacing with an error message and returning the local classifier prediction
Minor: added a logger instead of the `System.out` and `System.err` output
Minor: Updated the version of CommaSRL in the pipeline

**NOTE**: Cannot test this locally because of errors while downloading the Maven dependencies (e.g. datastore)

### CODE REVIEW

**Comment by mssammon** [comment]
@christos-c can you say more about the datastore errors? There is some maintenance going on this morning that might have disrupted the datastore server... @danyaljj fyi

**Comment by christos-c** [comment]
@mssammon it's not directly related to the store itself. Rather, it's the Maven dependency of the datastore package that doesn't seem to be able to be downloaded. This might not be related, but the semaphore execution failed with the message `execution expired` (while downloading Maven deps)

**Comment by danyaljj** [comment]
Recently we've been ignoring semaphore most and only trusting the results of Jenkins (which is unfortunately down due to outage).

File: pipeline/pom.xml [hunk 1]
```diff
@@ -50,7 +50,7 @@
         <dependency>
             <groupId>edu.illinois.cs.cogcomp</groupId>
             <artifactId>illinois-comma</artifactId>
-            <version>2.3.4</version>
+            <version>3.1.19</version>
```
Comment (danyaljj) [inline pipeline/pom.xml]: nice.

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/ConstrainedPrepSRLClassifier.java [hunk 2]
```diff
@@ -70,16 +71,16 @@ public String discreteValue(Object example) {
...
         } catch (Exception e) {
-            System.err
-                    .println("LBJava ERROR: Fatal error while evaluating classifier ConstrainedPrepSRLClassifier: "
+            logger.error
+                    ("LBJava ERROR: Fatal error while evaluating classifier ConstrainedPrepSRLClassifier: "
                             + e);
-            e.printStackTrace();
-            System.exit(1);
+            logger.error("Returning local classifier prediction.");
+            result = prepSRLClassifier.discreteValue(example);
```
Comment (danyaljj) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/ConstrainedPrepSRLClassifier.java]: cool.

**Comment by danyaljj** [comment]
Looking at the changes, it looks good to me to merge. Just to be on the safe side, we can wait a couple of hours so that Jenkins comes back up, so that we merge after a little testing.

**Comment by danyaljj** [comment]
Merging this. Will fix Jenkins issue later.

File: prepsrl/src/test/java/edu/illinois/cs/cogcomp/prepsrl/PrepSRLAnnotatorTest.java [hunk 3]
```diff
@@ -55,4 +55,15 @@ public void testLong() throws Exception {
...
+    @Test
+    public void testException() throws Exception {
+        String longSentence = "That put him a stroke ahead of Curt Byrum, John Adams and " +
+                "Lonnie Clements , all of whom are looking for their first " +
+                "victory on the PGA Tour .";
+        TextAnnotation ta = TextAnnotationUtilities.createFromTokenizedString(longSentence);
+        preprocessor.annotate(ta);
+        View view = annotator.getView(ta);
+        assertEquals("ahead of", view.getConstituents().get(0).getSurfaceForm());
+    }
```
Comment (danyaljj) [inline prepsrl/src/test/java/edu/illinois/cs/cogcomp/prepsrl/PrepSRLAnnotatorTest.java]: @christos-c does this pass on your machine?
In my machine it gives: 
```
https://lists.sourceforge.net/lists/listinfo/ojalgo-user
ILP ERROR: Unsatisfiable constraints!
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 42.886 sec <<< FAILURE! - in edu.illinois.cs.cogcomp.prepsrl.PrepSRLAnnotatorTest
testException(edu.illinois.cs.cogcomp.prepsrl.PrepSRLAnnotatorTest)  Time elapsed: 0.502 sec  <<< FAILURE!
java.lang.AssertionError: ILP: Constant encountered. (false)
	at edu.illinois.cs.cogcomp.lbjava.infer.ILPInference.visit(ILPInference.java:673)
	at edu.illinois.cs.cogcomp.lbjava.infer.PropositionalConstant.runVisit(PropositionalConstant.java:260)
	at edu.illinois.cs.cogcomp.lbjava.infer.ILPInference.infer(ILPInference.java:232)
	at edu.illinois.cs.cogcomp.lbjava.infer.ILPInference.valueOf(ILPInference.java:276)
	at edu.illinois.cs.cogcomp.prepsrl.inference.ConstrainedPrepSRLClassifier.discreteValue(ConstrainedPrepSRLClassifier.java:77)
	at edu.illinois.cs.cogcomp.prepsrl.PrepSRLAnnotator.addView(PrepSRLAnnotator.java:85)
	at edu.illinois.cs.cogcomp.annotation.Annotator.lazyAddView(Annotator.java:181)
	at edu.illinois.cs.cogcomp.annotation.Annotator.getView(Annotator.java:166)
	at edu.illinois.cs.cogcomp.prepsrl.PrepSRLAnnotatorTest.testException(PrepSRLAnnotatorTest.java:66)


Results :

Failed tests: 
  PrepSRLAnnotatorTest.testException:66 ILP: Constant encountered. (false)
```
I changed the catch statement to: 
```
} catch (Exception | AssertionError e) {
```
where it's failing due to existing span. Thoughts?

Comment (christos-c) [inline prepsrl/src/test/java/edu/illinois/cs/cogcomp/prepsrl/PrepSRLAnnotatorTest.java]: @danyaljj as I said, I haven't been able to run the tests due to the Maven packages errors, so I don't know why that exception is thrown -- maybe 'ahead of' is not the first constituent (or it's not recognised at all?). I'll try again now and see if I can get this to work on my local machine.

Comment (danyaljj) [inline prepsrl/src/test/java/edu/illinois/cs/cogcomp/prepsrl/PrepSRLAnnotatorTest.java]: The dependencies should work fine now (servers back up). Lmk if you find out what's wrong with the test.

**Comment by christos-c** [comment]
@danyaljj I fixed the error (it had to do with allowing overlapping annotations).
FYI I'm getting a non-fatal exception when loading some items from the datastore:
```
Downloading the file from datastore . . . 
		GroupId: readonly.org.cogcomp.roget.thesaurus
		ArtifactId: 1.3/rogetThesaurus
May 25, 2017 10:44:22 AM io.minio.http.HeaderParser set
SEVERE: exception occured: 
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at io.minio.http.HeaderParser.set(HeaderParser.java:73)
	at io.minio.MinioClient.execute(MinioClient.java:841)
	at io.minio.MinioClient.executeHead(MinioClient.java:1018)
	at io.minio.MinioClient.statObject(MinioClient.java:1154)
	at org.cogcomp.Datastore.getFile(Datastore.java:180)
	at org.cogcomp.Datastore.getFile(Datastore.java:149)
	at edu.illinois.cs.cogcomp.edison.features.factory.RogetThesaurusFeatures.loadFromDatastore(RogetThesaurusFeatures.java:67)
Caused by: java.lang.IllegalArgumentException: Invalid format: "Thu, 25 May 2017 15:44:22 GMT" is malformed at "GMT"
	at org.joda.time.format.DateTimeFormatter.parseDateTime(DateTimeFormatter.java:866)
	at io.minio.ResponseHeader.setDate(ResponseHeader.java:89)
```

**Comment by danyaljj** [comment]
Whichever gives the error, has to contain joda-time as dependency, like [this](https://github.com/CogComp/cogcomp-nlp/blob/f54cc9301558b98496156fc2a99ccb0016604c79/commasrl/pom.xml#L98-L103).

**Pull request merged**